coderd/users.go

```diff
@@ -200,6 +202,19 @@ func (api *API) postFirstUser(rw http.ResponseWriter, r *http.Request) {
 		return
 	}
 
+	//nolint:gocritic // Needed to create first user.
+	if _, err := api.Database.UpdateUserProfile(dbauthz.AsSystemRestricted(ctx), database.UpdateUserProfileParams{
```

Comment:
Why do we need to run a separate query to update the name? In my head, it would be done in the `CreateUserProfile`.

Reply:
I'm not sure what you mean; there is no such query `CreateUserProfile`. Do you mean `InsertUser`?

Reply:
Here: https://github.com/coder/coder/blob/3eadba7c95f92fe5011e35c9eb98df8709816b70/coderd/users.go#L188C2-L196C4

Reply:
I can't see a good reason to create a user without the name and after, update it. IMO, it should be a single operation.